### PR TITLE
added additional intrinsic functions

### PR DESCRIPTION
**Pull request by josh146**
Very small PR, just looks big due to autogenerated files.

### CODE REVIEW

File: doc/syntax.rst [hunk 1]
```diff
@@ -201,10 +201,12 @@ Functions
 
 Blackbird also supports the intrinsic functions
```
Comment (co9olguy) [inline doc/syntax.rst]: I feel like the name "intrinsic functions" is not clear to users

File: blackbird_python/blackbird/tests/test_auxiliary.py [hunk 2]
```diff
@@ -181,6 +181,14 @@ def test_function_exp(self, parser, ctx, n, expected, num):
         expression = num(n)
         assert _func(func, expression) == np.exp(expected)
 
+    @pytest.mark.parametrize('n, expected', test_complex)
+    def test_function_log(self, parser, ctx, n, expected, num):
+        """Test that a Blackbird log function is properly called"""
+        func = blackbirdParser.FunctionContext(parser, ctx)
+        func.LOG = lambda: True
```
Comment (co9olguy) [inline blackbird_python/blackbird/tests/test_auxiliary.py]: just curious, what is the role of these lambdas?

File: blackbird_python/blackbird/tests/test_auxiliary.py [hunk 3]
```diff
@@ -197,6 +205,96 @@ def test_function_cos(self, parser, ctx, n, expected, num):
...
+    def test_function_tan(self, parser, ctx, n, expected, num):
+        """Test that a Blackbird tan function is properly called"""
+        func = blackbirdParser.FunctionContext(parser, ctx)
+        func.TAN = lambda: True
+        expression = num(n, num_type='float')
+        assert _func(func, expression) == np.tan(expected)
+
+    def test_function_arcsin(self, parser, ctx, num):
+        """Test that a Blackbird arcsin function is properly called"""
+        n = "0.543"
```
Comment (co9olguy) [inline blackbird_python/blackbird/tests/test_auxiliary.py]: How come some of these are tested for multiple values and others for a single value?

File: blackbird_python/blackbird/blackbirdListener.py [hunk 4]
```diff
@@ -242,30 +242,21 @@ def exitModes(self, ctx:blackbirdParser.ModesContext):
...
-    # Enter a parse tree produced by blackbirdParser#MulLabel.
-    def enterMulLabel(self, ctx:blackbirdParser.MulLabelContext):
+    # Enter a parse tree produced by blackbirdParser#BracketsLabel.
+    def enterBracketsLabel(self, ctx:blackbirdParser.BracketsLabelContext):
         pass
 
-    # Exit a parse tree produced by blackbirdParser#MulLabel.
-    def exitMulLabel(self, ctx:blackbirdParser.MulLabelContext):
+    # Exit a parse tree produced by blackbirdParser#BracketsLabel.
+    def exitBracketsLabel(self, ctx:blackbirdParser.BracketsLabelContext):
```
Comment (co9olguy) [inline blackbird_python/blackbird/blackbirdListener.py]: I think(?) these are autogenerated, but just curious: what changed?

File: blackbird_python/blackbird/tests/test_auxiliary.py (hunk 2 above)
Comment (josh146) [reply blackbird_python/blackbird/tests/test_auxiliary.py]: The way the parser works, if it detects a function, it produces a `FunctionContext` object with various method, each method corresponding to the function type.

So if `log(0.54)` is in the Blackbird code, the parser creates a `func = FunctionContext()` object with `func.LOG()` returning `True`.

Comment (co9olguy) [reply blackbird_python/blackbird/tests/test_auxiliary.py]: :thinking:

File: blackbird_python/blackbird/tests/test_auxiliary.py (hunk 3 above)
Comment (josh146) [reply blackbird_python/blackbird/tests/test_auxiliary.py]: Some of the functions are very picky with allowed values, and I suddenly thought that using huge amounts of test data isn't necessary - a single value passing should indicate it is the correct function.

File: blackbird_python/blackbird/blackbirdListener.py (hunk 4 above)
Comment (josh146) [reply blackbird_python/blackbird/blackbirdListener.py]: I think everything got shifted down in the autogenerated files to make room for the new functions!

File: blackbird_python/blackbird/tests/test_auxiliary.py (hunk 3 above)
Comment (josh146) [reply blackbird_python/blackbird/tests/test_auxiliary.py]: I think I will standardize this in a new PR.

Comment (co9olguy) [reply blackbird_python/blackbird/tests/test_auxiliary.py]: :+1:

**Pull request merged**